### PR TITLE
[Validator] Html5 Email Validation

### PR DESCRIPTION
**Pull request by PurpleBooth**
Add documentation about the new mode parameter. Adds descriptions for
the 'loose', 'strict', and 'html5' options.

Relates to symfony/symfony#24442

### CODE REVIEW

**Comment by PurpleBooth** [comment]
I've rebased this off master, as eventually the PR went into the 4.1 branch

File: reference/constraints/Email.rst [hunk 1]
```diff
@@ -91,14 +91,34 @@ Basic Usage
...
+
+A simple regular expression. Allows all values with an "@" symbol in, and a "."
+in the second host part of the email address.
+
 strict
-~~~~~~
+......
 
-**type**: ``boolean`` **default**: ``false``
+Uses the `egulias/email-validator`_ library is to perform an RFC compliant
```
Comment (xabbuh) [inline reference/constraints/Email.rst]: [...] library ~is~to perform [...]

File: reference/constraints/Email.rst [hunk 2]
```diff
@@ -91,14 +91,34 @@ Basic Usage
 Options
 -------
 
+mode
+~~~~
+
+**type**: ``string`` **default**: ``loose``
+
+This option is optional and pattern the email address is validated against.
```
Comment (xabbuh) [inline reference/constraints/Email.rst]: This option is optional and defines the pattern the e-mail address is validated against.

File: reference/constraints/Email.rst [hunk 3]
```diff
@@ -91,14 +91,34 @@ Basic Usage
...
+
+* ``loose``
+* ``strict``
+* ``html5``
+
+loose
+.....
+
+A simple regular expression. Allows all values with an "@" symbol in, and a "."
+in the second host part of the email address.
```
Comment (xabbuh) [inline reference/constraints/Email.rst]: e-mail

**Review by xabbuh — APPROVED** [review]
This looks great. I have just left a few minor comments. Thanks for creating the feature and writing the documentation for it.

File: reference/constraints/Email.rst (hunk 3 above)
Comment (PurpleBooth) [reply reference/constraints/Email.rst]: I had a look across the other documentation as to what we tend to use, and it seems to be "email".

I've switched everything to email in this document, plus the other email related document, so we're consistent with everything else.

**Comment by xabbuh** [comment]
Thank you @PurpleBooth.

File: reference/constraints/Email.rst [hunk 4]
```diff
@@ -7,7 +7,7 @@ cast to a string before being validated.
 +----------------+---------------------------------------------------------------------+
 | Applies to     | :ref:`property or method <validation-property-target>`              |
 +----------------+---------------------------------------------------------------------+
-| Options        | - `strict`_                                                         |
+| Options        | - `mode`_                                                           |
```
Comment (guilliamxavier) [inline reference/constraints/Email.rst]: Shouldn't the `strict` option have been annotated as "deprecated in Symfony 4.1" rather than completely replaced?

**Pull request merged**